enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SelectEmbedOptionsStep.tsx

```diff
@@ -35,48 +46,52 @@ export const SelectEmbedOptionsStep = () => {
 
   return (
     <Stack gap="md">
-      <Card p="md">
-        <Text size="lg" fw="bold" mb="md">
-          {t`Behavior`}
-        </Text>
-        <Stack gap="md">
-          {isQuestionOrDashboardEmbed && (
-            <Checkbox
-              label={t`Allow people to drill through on data points`}
-              checked={settings.drills}
-              onChange={(e) => updateSettings({ drills: e.target.checked })}
-            />
-          )}
-
-          {isDashboardOrQuestion && (
-            <Checkbox
-              label={t`Allow downloads`}
-              checked={settings.withDownloads}
-              onChange={(e) =>
-                updateSettings({ withDownloads: e.target.checked })
-              }
-            />
-          )}
-
-          {isQuestionComponent && (
-            <Checkbox
-              label={t`Allow people to save new questions`}
-              checked={settings.isSaveEnabled}
-              onChange={(e) =>
-                updateSettings({ isSaveEnabled: e.target.checked })
-              }
-            />
-          )}
-
-          {isBrowserComponent && (
-            <Checkbox
-              label={t`Allow editing dashboards and questions`}
-              checked={!settings.readOnly}
-              onChange={(e) => updateSettings({ readOnly: !e.target.checked })}
-            />
-          )}
-        </Stack>
-      </Card>
+      {!isMetabotComponent && (
```

Comment:
can we do something about all these boolean flags? I don't think this is sustainable. Adding one or 2 new components, and this file will be spinkles with 5 or 6 more flags most likely. I don't even think it's that readable at the moment.

Reply:
Fixed in https://github.com/metabase/metabase/pull/64388/commits/4b5b2cd231f4b587f25d7cb64f1789c530b25784 and https://github.com/metabase/metabase/pull/64388/commits/79935ff0f0149679372cd83bd4c015a3728717c5, used ts-pattern to make it more explicit.